### PR TITLE
feat(cli): early protection for invalid prometheus prefix

### PR DESCRIPTION
**Pull request by jansegre**
### Motivation

We had a report from someone accidentally using an invalid prefix through `HATHOR_PROMETHEUS_PREFIX` which had a message that was hard to understand.

### Acceptance Criteria

- Check for invalid Prometheus prefix when building and give a more helpful message.

### Checklist

- [x] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged

### CODE REVIEW

**Review by msbrogli — APPROVED** [review]

File: hathor/builder/resources_builder.py [hunk 1]
```diff
@@ -60,9 +86,14 @@ def build(self) -> Optional[server.Site]:
         return None
 
     def create_prometheus(self) -> PrometheusMetricsExporter:
+        prometheus_prefix = self._args.prometheus_prefix
+        if self._args.prometheus_prefix and is_prometheus_metric_name_valid(prometheus_prefix):
```
Comment (glevco) [inline hathor/builder/resources_builder.py:90]: Missing `not`?

**Review by glevco — CHANGES_REQUESTED** [review]

File: hathor/builder/resources_builder.py (hunk 1 above)
Comment (jansegre) [reply hathor/builder/resources_builder.py:90]: True! I missed it after refactoring.

Comment (jansegre) [reply hathor/builder/resources_builder.py]: Fixed.

**Review by glevco — APPROVED** [review]

File: hathor/builder/resources_builder.py (hunk 1 above)
Comment (glevco) [reply hathor/builder/resources_builder.py]: Thanks!

**Review by msbrogli — APPROVED** [review]

**Pull request merged**